buildSrc/src/main/kotlin/com/datadog/gradle/plugin/transdeps/TransitiveDependenciesPlugin.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.gradle.plugin.transdeps
+
+import com.datadog.gradle.config.taskConfig
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class TransitiveDependenciesPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+
+        val task = target.tasks.create(TASK_NAME, TransitiveDependenciesTask::class.java)
+        task.outputFile = File(target.projectDir, FILE_NAME)
+
+        target.taskConfig<KotlinCompile> {
```

Comment:
I would rather run this task every time we do a new release. We can just bundle it with the `publishToMaven` task instead of running it all the time. Our build time is already taking some time so let's try to optimise it a bit.

Reply:
The thing is, if we run it only on the CI we can't update the file in git

Reply:
Damn 🤔